clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -720,14 +736,30 @@ public FetchResponseMetricAggregator(FetchManagerMetrics sensors,
          * and number of records parsed. After all partitions have reported, we write the metric.
          */
         public void record(TopicPartition partition, int bytes, int records) {
-            unrecordedPartitions.remove(partition);
-            totalBytes += bytes;
-            totalRecords += records;
-
-            if (unrecordedPartitions.isEmpty()) {
+            this.unrecordedPartitions.remove(partition);
+            this.fetchMetrics.fetchBytes += bytes;
```

Comment:
Minor: we could turn this and the line below into a single method call, maybe `FetchMetrics.increment(bytes, records)`?